src/api/spec/controllers/webui/project_controller_spec.rb

```diff
@@ -687,4 +691,34 @@
       is_expected.to redirect_to(project_show_path(user.home_project))
     end
   end
+
+  describe 'POST #new_incident_request' do
+    before do
+      login user
+      request.env["HTTP_REFERER"] = root_url # Needed for the redirect_to :back
+    end
+
+    it "without an existent project will raise an exception" do
+      expect { post :new_incident_request, project: 'non:existent:project' }.to raise_error Project::UnknownObjectError
+    end
+
+    context "without a proper action for the maintenance project" do
+      before do
+        post :new_incident_request, project: maintenance_project, description: "Fake description for a request"
+      end
+
+      it { expect(flash[:error]).to eq("MaintenanceHelper::MissingAction") }
+      it { is_expected.to redirect_to(:back) }
+    end
+
+    context "with the proper params" do
+      before do
+        BsRequest.any_instance.stubs(:save!).returns(true)
+        post :new_incident_request, project: maintenance_project, description: "Fake description for a request"
```

Comment:
Because then I will end up having to create too much stuff in the backend just for testing the default case.

I mean,  I'm testing the workflow of the action (the normal flow in this case), not the `save!` method of BsRequest. 

So, if I want to have this working using **real** data I need a package which has a _link file pointing to a project which is defined as maintained project in the maintenance project... looks like too complicated just for testing the  case in that `save!` doesn't throw any of those exceptions.